packages/schema/src/interfaces.ts

```diff
@@ -232,6 +233,9 @@ export interface IJupyterGISModel extends DocumentRegistry.IModel {
   triggerLayerUpdate(layerId: string, layer: IJGISLayer): void;
 
   disposed: ISignal<any, void>;
+  isDrawVectorLayerEnabled: boolean;
+  updateIsDrawVectorLayerEnabled(): void;
+  selectedVectorLayerSourceId: string;
```

Comment:
Introducing this would require us to make sure that value is always up-to-date, which is unneeded added complexity.

I don't think we need to introduce this, we already have a way to get the selected layer (then its source) from the main view:

```typescript
const localState = this._model?.sharedModel.awareness.getLocalState();
const selectedLayer = localState?.selected?.value;
```

Reply:
@martinRenou Thanks for the comment. I will change this.